src/datasets/formatting/__init__.py

```diff
@@ -70,32 +77,32 @@ def _register_unavailable_formatter(
 
 
 # Here we define all the available formatting functions that can be used by `Dataset.set_format`
-_register_formatter(PythonFormatter, None, aliases=["python"])
+_register_formatter(PythonFormatter, None, aliases=["python"], supports_lazy_formatting=True)
```

Comment:
Maybe supports_lazy_formatting can be a class attribute instead ? Since this is implementation - specific

Reply:
Only the default formatter (Python) can be lazily decoded now, so I removed this `supports_lazy_formatting` API and added `lazy` as a param to the `PythonFormatter`. If we implement lazy decoding for more formatters in the future, we can come up with a more generic approach.